lib/runner/cli/cli.js

```diff
@@ -354,7 +372,11 @@ class CliRunner {
         newConfigCreated = CliRunner.createDefaultConfig(localJsOrTsValue);
       }
 
-      if (hasJsOrTsConfig || newConfigCreated) {
+      if (Utils.fileExistsSync(CliRunner.CONFIG_FILE_TS)) {
```

Comment:
We should ideally use `localJsOrTsValue` here as our source of truth to decide if the ts config file is our primary config for the project. This is because the logic used to find `localJsOrTsValue` may change later and we may give preference to some other config file even if a `nightwatch.conf.ts` file is present.

This could be solved by changing this statement to:
`localJsOrTsValue.includes(path.basename(CliRunner.CONFIG_FILE_TS))`

And for the issue you were facing with tests, that could be solved by changing the `basename` definition in mockery to:
```
    basename(a) {
        if (a === './globals.json') {
          return 'globals';
        }

        return origPath.basename(a);
      },
```